test/components/views/settings/tabs/room/RolesRoomSettingsTab-test.tsx

```diff
@@ -29,23 +32,52 @@ import { ElementCall } from "../../../../../../src/models/Call";
 describe("RolesRoomSettingsTab", () => {
     const roomId = "!room:example.com";
     let cli: MatrixClient;
+    let room: Room;
 
     const renderTab = (): RenderResult => {
         return render(<RolesRoomSettingsTab roomId={roomId} />);
     };
 
-    const getVoiceBroadcastsSelect = () => {
+    const getVoiceBroadcastsSelect = (): HTMLElement => {
         return renderTab().container.querySelector("select[label='Voice broadcasts']");
     };
 
-    const getVoiceBroadcastsSelectedOption = () => {
+    const getVoiceBroadcastsSelectedOption = (): HTMLElement => {
         return renderTab().container.querySelector("select[label='Voice broadcasts'] option:checked");
     };
 
     beforeEach(() => {
         stubClient();
         cli = MatrixClientPeg.get();
-        mkStubRoom(roomId, "test room", cli);
+        room = mkStubRoom(roomId, "test room", cli);
+    });
+
+    it("should allow an Admin to demote themselves but not others", () => {
```

Comment:
Where do we test for the first case? Am I blind?

Reply:
https://github.com/matrix-org/matrix-react-sdk/pull/9681/files#diff-112b8dbf45c8c57a344b4e60001a330a13cc82af0019c1f8fed90d91bf0eabf8R79